src/Petabridge.Collections/CircularBuffer.cs

```diff
@@ -35,10 +35,28 @@ public class CircularBuffer<T> : ICircularBuffer<T>
         protected T[] Buffer;
 
         public CircularBuffer(int capacity)
+        {
+            if (capacity > 0)
+            {
+                _head = 0;
+                _tail = 0;
+                Buffer = new T[capacity];
+            }
+            else
+            {
+                throw new Exception("Buffer size has to be greater than 0");
+            }
+        }
+
+        public CircularBuffer(IEnumerable<T> values, int capacity)
         {
             _head = 0;
             _tail = 0;
             Buffer = new T[capacity];
+            foreach (var x in values)
+            {
+                Add(x);
```

Comment:
Yeah, that's probably the simplest way to do this for the time being without duplicating the circular buffer's logic.